source/agora/test/GetEnrollments.d

```diff
@@ -0,0 +1,137 @@
+/*******************************************************************************
+
+    Contains tests for getting missing enrollments from the network
+
+    Copyright:
+        Copyright (c) 2019-2021 BOSAGORA Foundation
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.test.GetEnrollments;
+
+version (unittest):
+
+import agora.api.FullNode;
+import agora.common.Amount;
+import agora.common.Config;
+import agora.consensus.data.Block;
+import agora.consensus.data.Enrollment;
+import agora.consensus.data.Transaction;
+import agora.crypto.Key;
+import agora.test.Base;
+
+import core.atomic : atomicLoad;
+import core.thread;
+
+/// Situation: There are six validators enrolled in Genesis block.
+///     But a validator can not receive any enrollments in the case
+///     of an abnormal situation. The validator tries to get missing
+///     enrollments periodically
+/// Expectation: The validator ends up restoring misssing enrollments
+///     from other nodes.
+unittest
+{
+    static class CustomValidator : TestValidatorNode
+    {
+        mixin ForwardCtor!();
+
+        private shared bool* enable_catchup;
+
+        ///
+        public this (Parameters!(TestValidatorNode.__ctor) args,
+            shared(bool)* enable_catchup)
+        {
+            this.enable_catchup = enable_catchup;
+            super(args);
+        }
+
+        ///
+        protected override void catchupTask () nothrow
+        {
+            if (atomicLoad(*this.enable_catchup))
+                super.catchupTask();
+        }
```

Comment:
Why do we override `catchupTask` and not `catchupEnrollment` ?

Reply:
It's because I wanna make the node one block behind the latest height.